Taarafo.Core.Tests.Unit/Services/Foundations/Profiles/ProfileServiceTests.Exceptions.RetrieveAll.cs

```diff
@@ -0,0 +1,91 @@
+﻿// ---------------------------------------------------------------
+// Copyright (c) Coalition of the Good-Hearted Engineers
+// FREE TO USE TO CONNECT THE WORLD
+// ---------------------------------------------------------------
+
+using System;
+using System.Threading.Tasks;
+using Microsoft.Data.SqlClient;
+using Moq;
+using Taarafo.Core.Models.Profiles.Exceptions;
+using Xunit;
+
+namespace Taarafo.Core.Tests.Unit.Services.Foundations.Profiles
+{
+    public partial class ProfileServiceTests
+    {
+        [Fact]
+        public void ShouldThrowCriticalDependencyExceptionOnRetrieveAllWhenSqlExceptionOccursAndLogIt()
+        {
+            // given
+            SqlException sqlException = GetSqlException();
+
+            var failedProfileStorageException =
+                new FailedProfileStorageException(sqlException);
+
+            var expectedProfileDependencyException = 
+                new ProfileDependencyException(failedProfileStorageException);
+
+            this.storageBrokerMock.Setup(broker =>
+                broker.SelectAllProfiles())
+                    .Throws(sqlException);
+
+            // when
+            Action retrieveAllProfileAction = () =>
+                this.profileService.RetrieveAllProfiles();
+
+            // then
+            Assert.Throws<ProfileDependencyException>(
+                retrieveAllProfileAction);
+
+            this.storageBrokerMock.Verify(broker =>
+                broker.SelectAllProfiles());
+
+            this.loggingBrokerMock.Verify(broker =>
+                broker.LogCritical(It.Is(SameExceptionAs(
+                    expectedProfileDependencyException))),
+                        Times.Once);
+
+            this.storageBrokerMock.VerifyNoOtherCalls();
+            this.loggingBrokerMock.VerifyNoOtherCalls();
```

Comment:
`dateTimeBroker` validates are missing here.

Reply:
dateTimeBroker was added for validation